docs/performance/inserts/architecture.txt

```diff
@@ -0,0 +1,250 @@
+=================
+Insert Processing
+=================
+
+To optimize CrateDB and your client applications for insert performance, it's
+important that you have a basic understanding of how CrateDB handles inserts
+internally. This will make it easier to reason about the location of potential
+performance bottlenecks.
+
+.. rubric:: Table of Contents
+
+.. contents::
+   :local:
+
+Types of Insert
+===============
+
+There are three kinds of inserts as far as CrateDB's insert handling is
+concerned.
+
+Single Inserts
+--------------
+
+A single insert looks like this:
+
+.. code-block:: psql
+
+  cr> INSERT INTO my_table (id, first_name)
+  ...      VALUES (1, "Arthur");
+  INSERT OK, 1 row affected  (... sec)
+
+As far as CrateDB's insert handling is concerned,
```

Comment:
As discussed jdbc prep statements batch insert should be mentioned in the Bulk inserts.
It's treated in the execution as an HTTP Bulk insert, only overhead is extra network communication on .addBatch() calls